### PR TITLE
kanidm: require big-parallel for building

### PR DESCRIPTION
**Pull request by adamcstephens**
## Description of changes

Build has timed out and been killed multiple times on hydra builds. This forces users to attempt to build it themselves, which they cannot due without sufficient resources.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ka/kanidm/package.nix [hunk 1]
```diff
@@ -78,6 +78,9 @@ rustPlatform.buildRustPackage rec {
     rust-jemalloc-sys
   ];
 
+  # can take over 4 hours on 2 cores and needs 16GB+ RAM
+  requiredSystemFeatures = [ "big-parallel" ];
+
```
Comment (Flakebi) [inline pkgs/by-name/ka/kanidm/package.nix:83]: nitpick: In most packages, this is placed before `passthru` or `meta`, currently it’s a bit in the middle between build-recipe related things.

**Review by Flakebi — APPROVED** [review]
Sounds good to me

File: pkgs/by-name/ka/kanidm/package.nix (hunk 1 above)
Comment (adamcstephens) [reply pkgs/by-name/ka/kanidm/package.nix]: Cool, moved. It seemed build related to me, but I'm not picky on the location.

**Comment by erictapen** [comment]
A bit offtopic, but how would somebody apply for job restart privileges on hydra.nixos.org? I noticed the build failure early and would be available in the future to restart the job.

**Comment by adamcstephens** [comment]
I'm not sure the official process. I asked in the matrix infra room and was given the permission.

**Pull request merged**